pkg/scheduler/api/resource_info.go

```diff
@@ -141,22 +141,17 @@ func (r *Resource) Add(rr *Resource) *Resource {
 
 //Sub subtracts two Resource objects.
 func (r *Resource) Sub(rr *Resource) *Resource {
-	if rr.LessEqual(r) {
-		r.MilliCPU -= rr.MilliCPU
-		r.Memory -= rr.Memory
+	r.MilliCPU -= rr.MilliCPU
+	r.Memory -= rr.Memory
 
-		for rrName, rrQuant := range rr.ScalarResources {
-			if r.ScalarResources == nil {
-				return r
-			}
-			r.ScalarResources[rrName] -= rrQuant
+	for rrName, rrQuant := range rr.ScalarResources {
+		if r.ScalarResources == nil {
+			return r
 		}
-
-		return r
+		r.ScalarResources[rrName] -= rrQuant
 	}
 
-	panic(fmt.Errorf("Resource is not sufficient to do operation: <%v> sub <%v>",
```

Comment:
keep panic

Reply:
by design negative resource is now possible when resource is allocated using backfill job resources.

Reply:
that's in-correct to me; I think we discussed this before, we need to keep panic; if negative integer is necessary, add other field to handle that.

Reply:
"if negative integer is necessary, add other field to handle that" we should allow it. why bother with another field?

Reply:
keep panic, it catches several issue there.

Reply:
btw, please get confirm before resolve conversation.